ruma-api-macros/src/api/response.rs

```diff
@@ -123,18 +123,20 @@ impl Response {
                     {
                         quote! {
                             if let Some(header) = response.#field_name {
-                                resp_builder = resp_builder.header(
-                                    #import_path::exports::http::header::#header_name,
-                                    header,
-                                );
+                                headers
+                                    .insert(
+                                        #import_path::exports::http::header::#header_name,
+                                        header.parse().unwrap(),
```

Comment:
I think we were handling invalid header values without panicking before, right?

Reply:
Sort of, it was just done implicitly https://docs.rs/http/0.2.1/src/http/response.rs.html#626 and then we would have panicked when we unwrap to the response from the builder.

Reply:
Maybe adding a more direct macro error if we have anything that isn't a `String`, now we get an error about the type not being able to parse? Since we can't test the header values for being valid at macro time.

Reply:
If you have a look at [ruma_api/error.rs](https://github.com/ruma/ruma/blob/60bd16ccfc340758f0c700ba265759e52c37711c/ruma-api/src/error.rs#L210), you'll see we already have HTTP header conversion variants for both serialization and deserialization (just search for `http::header`). I think we can just do the same error handling we do in the other parts of the request / response conversion code.

Reply:
I went for the `?` directly since there is a `From` impl for `http::header::InvalidHeaderValue` error. We could also remove the unwrap on `let response = resp_builder.body(#body).unwrap();` with `?`, although now there is no way this could panic since it would be caught by the `parse` error first.